File: sample-api/src/main/java/Product.java
import com.google.gson.annotations.SerializedName;

public class Product {

  @SerializedName("product_name")
  public final String name;

  @SerializedName("price")
  public final String price;

  @SerializedName("id")
  public final String id;

  public Product(String name, String price, String id){
    this.name = name;
    this.price = price;
    this.id = id;
  }

}

File: android-client/app/src/main/java/com/abnormallydriven/paginglibrarytestdrive/products/Product.kt
package com.abnormallydriven.paginglibrarytestdrive.products

import com.google.gson.annotations.SerializedName


class Product(@field:SerializedName("product_name")
              val name: String,

              @field:SerializedName("price")
              val price: String,

              @field:SerializedName("id")
              val id: String)